ocs_ci/ocs/ocp.py

```diff
@@ -1094,15 +1095,112 @@ def verify_cluster_operator_status(cluster_operator):
+        elif condition['type'] == 'Progressing' and condition['status'] == 'True':
             log.info(f'{cluster_operator} status is Progressing')
             return False
     log.info(f'{cluster_operator} status is valid')
 
     return True
+
+
+def validate_cluster_version_status():
+    """
+    Verify OCP upgrade is completed, by checking 'oc get clusterversion'
+    status
+
+    Returns:
+        bool: False in case that one of condition flags is invalid:
+            Progressing (should be False), Failing(should be False)
+            or Available (should be True)
+
+    """
+    ocp = OCP(kind="clusterversion")
+    operator_data = ocp.get('-o json', out_yaml_format=False)
+    conditions = operator_data['items'][0].get('status').get('conditions', [])
+    for condition in conditions:
+        if condition['type'] == 'Progressing' and condition['status'] == 'True':
+            log.info('cluster version status is Progressing')
+            return False
+        elif condition['type'] == 'Failing' and condition['status'] == 'True':
+            log.info('cluster version status is Failing')
+            return False
+        elif condition['type'] == 'Available' and condition['status'] != 'True':
+            log.info('cluster status is not available')
+            return False
+
+    log.info('Cluster version validation - OK!')
+    return True
+
+
+def get_ocp_upgrade_channel():
+    """
+    Gets OCP upgrade channel
+
+    Returns:
+        str: OCP upgrade channel name
+
+    """
+    ocp = OCP(kind="clusterversion")
+    log.info("Gathering Subscription Channel information")
+    operator_version = ocp.get('-o json', out_yaml_format=False)
+    log.debug(f"cluster version: {operator_version}")
+    channel = operator_version['items'][0].get('spec').get('channel')
+    log.info(f"Subscription Channel: {channel}")
+
+    return channel
+
+
+def patch_ocp_upgrade_channel(
+    channel_variable=config.UPGRADE['ocp_channel']
+):
+    """
+    Using 'oc patch clusterversion' if new OCP upgrade channel is
+    different than current one
+
+    Args:
+        channel_variable (str): New OCP upgrade subscription channel
+
+    """
+    if get_ocp_upgrade_channel() != channel_variable:
+        cmd = (
+            f'patch clusterversions/version -p \'{{"spec":'
+            f'{{"channel":"{channel_variable}"}}}}\' --type=merge'
+        )
+        ocp = OCP()
+        log.info(f"Patching channel into {channel_variable}")
+        ocp.exec_oc_cmd(cmd)
+
+    else:
+        log.info("No patch needed")
+
+
+def verify_ocp_upgrade_channel(
+    channel_variable=config.UPGRADE['ocp_channel']
+):
+    """
+    When upgrade OCP version, verify that subscription channel is same
+    as current one
+
+    Args:
+        channel_variable (str): New OCP upgrade subscription channel
+
+    Returns:
+        bool: True when OCP subscription channel is correct,
+        and no patch needed
```

Comment:
One indent right

Reply:
@ebenahar there was one unresolved comment in documentation. Couldn't this wait till fixed? @AviadP can you please fix in follow up PR?

Reply:
sure, will do that tomorrow morning